### PR TITLE
fix: show edit artwork button

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-1209]

### Description

This PR fixes the missing edit button on my collection artwork screen. I also added a bunch of tests to avoid the same regression in the future
| Before | After |
|--------|--------|
|  |  |

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- show edit artwork button - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1209]: https://artsyproduct.atlassian.net/browse/ONYX-1209?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (show edit artwork button - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8a561d09438d3460962ae7c8deff3563c53c4075
</p>

**Review by olerichter00 — COMMENTED** [review]
I think hiding the edit button whenever the artwork has an associated submission is what we want. We could consider allowing to edit the artwork when the submission is sold or rejected, but let's have a discussion with the team first ([Slack thread](https://artsy.slack.com/archives/C05EQL4R5N0/p1724228321667869?thread_ts=1724227182.859219&cid=C05EQL4R5N0) 🙏 ).

File: src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtwork.tsx [hunk 1]
```diff
@@ -87,12 +87,14 @@ const MyCollectionArtwork: React.FC<MyCollectionArtworkScreenProps> = ({
 
   const articles = extractNodes(artwork.artist?.articles)
 
+  const isEditable = !!artwork.consignmentSubmission?.internalID
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtwork.tsx:90]: 👀 it seems that there's an extra `!` there 😅 
No worries though, I can create a follow-up PR on it :)

File: src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtwork.tests.tsx [hunk 2]
```diff
@@ -18,6 +19,48 @@ describe("My Collection Artwork", () => {
...
+        mockEnvironment
+      )
+
+      resolveMostRecentRelayOperation(mockEnvironment, {
+        Artwork: () => ({
+          consignmentSubmission: null,
+        }),
+      })
+      expect(() => screen.getByText("Edit")).toThrow()
+    })
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtwork.tests.tsx:64]: 👏👏👏

**Pull request merged**